libp2p-identify/regen_structs_proto.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# This script regenerates the `src/structs_proto.rs` and `src/keys_proto.rs` files from
+# `structs.proto` and `keys.proto`.
+
+sudo docker run --rm -v `pwd`:/usr/code:z -w /usr/code rust /bin/bash -c " \
+    apt-get update; \
+    apt-get install -y protobuf-compiler; \
+    cargo install protobuf; \
+    protoc --rust_out . structs.proto"
+
+mv -f structs.rs ./src/structs_proto.rs
```

Comment:
Should this be a build script?

Reply:
That would require to have `protobuf` installed (both the software from Google and the Rust crate) on your machine, so IMO it's preferable to do it separately.

Reply:
You could have a `structs.proto.source` or whatever that is a clone of `structs.proto` at generation time and the build script can diff `structs.proto.source` and `structs.proto` to decide whether to regen `src/structs_proto.rs`. I'm just wondering whether at some point down the line we'll forget that this needs to be done so the earlier we automate this the less likely it is to cause problems.